### DISCUSSION TITLE
Routing All Traffic Through SOCKS Proxy

### DISCUSSION THREAD

**Discussion by NeverL4nd2** [shadowsocks/shadowsocks-rust#1715] [Q&A]
I have a socks proxy running at port 9960.
how can I use shadowsocks to route all network traffic from the host through the proxy?

**Comment by zonyitoo** [comment]
First of all, `sslocal` can handle data from the local network. So the key issue is that how to make `sslocal`'s outbound set to a socks proxy.

`sslocal` doesn't support socks outbound natively, you may try:

1. use a tool like proxychains to hook `sslocal`'s socket calls
2. make a Plugin for `sslocal` to send data through proxy to the destinated `ssserver`.
